internal/gossiper/target.go

```diff
@@ -252,111 +243,115 @@ func (g *Target[T]) HandleAppGossip(ctx context.Context, nodeID ids.NodeID, msg
 
 			// Check if we are going to propose if it has been less than
 			// [VerifyTimeout] since the last time we verified a block.
-			if time.Now().UnixMilli()-g.latestVerifiedTimestamp < g.cfg.VerifyTimeout {
-				proposers, err := g.validatorSet.Proposers(
+			if time.Now().UnixMilli()-t.latestVerifiedTimestamp < t.cfg.VerifyTimeout {
+				proposers, err := t.validatorSet.Proposers(
 					tctx,
-					g.cfg.NoGossipBuilderDiff,
+					t.cfg.NoGossipBuilderDiff,
 					1,
 				)
-				if err == nil && proposers.Contains(g.validatorSet.NodeID()) {
-					g.Queue(tctx) // requeue later in case peer validator
-					g.log.Debug("not gossiping because soon to propose")
+				if err == nil && proposers.Contains(t.validatorSet.NodeID()) {
+					t.Queue(tctx) // requeue later in case peer validator
+					t.log.Debug("not gossiping because soon to propose")
 					continue
 				} else if err != nil {
-					g.log.Warn("unable to determine if will propose soon, gossiping anyways", zap.Error(err))
+					t.log.Warn("unable to determine if will propose soon, gossiping anyways", zap.Error(err))
 				}
 			}
 
 			// Gossip to targeted nodes
-			if err := g.Force(tctx); err != nil {
-				g.log.Warn("gossip txs failed", zap.Error(err))
+			if err := t.Force(tctx); err != nil {
+				t.log.Warn("gossip txs failed", zap.Error(err))
 				continue
 			}
-		case <-g.stop:
-			g.log.Info("stopping gossip loop")
+		case <-t.stop:
+			t.log.Info("stopping gossip loop")
 			return
 		}
 	}
 }
 
-func (g *Target[T]) BlockVerified(t int64) {
-	if t < g.latestVerifiedTimestamp {
+func (t *Target[T]) BlockVerified(verifiedTimestamp int64) {
+	if verifiedTimestamp < t.latestVerifiedTimestamp {
 		return
 	}
-	g.latestVerifiedTimestamp = t
+	t.latestVerifiedTimestamp = verifiedTimestamp
 }
 
-func (g *Target[T]) Done() {
-	g.timer.Stop()
-	<-g.doneGossip
+func (t *Target[T]) Done() {
+	t.timer.Stop()
+	<-t.doneGossip
 }
 
-func (g *Target[T]) sendTxs(ctx context.Context, txs []T) error {
-	ctx, span := g.tracer.Start(ctx, "Gossiper.sendTxs")
+func (t *Target[T]) sendTxs(ctx context.Context, txs []T) error {
+	ctx, span := t.tracer.Start(ctx, "Gossiper.sendTxs")
 	defer span.End()
 
-	gossipContainers, err := g.targetStrategy.Target(ctx, txs)
+	start := time.Now()
+	gossipContainers, err := t.targetStrategy.Target(ctx, txs)
 	if err != nil {
 		return err
 	}
 
+	numTxsGossipped := 0
 	for _, gossipContainer := range gossipContainers {
-		// Marshal gossip
-		b, err := g.serializer.Marshal(gossipContainer.Txs)
+		numTxsGossipped += len(gossipContainer.Txs)
+		b, err := t.serializer.Marshal(gossipContainer.Txs)
 		if err != nil {
 			return err
 		}
 
 		// Send gossip to specified peers
-		if err := g.client.AppGossip(ctx, common.SendConfig{NodeIDs: gossipContainer.NodeIDs}, b); err != nil {
+		if err := t.client.AppGossip(ctx, common.SendConfig{NodeIDs: gossipContainer.NodeIDs}, b); err != nil {
 			return err
 		}
 	}
+	t.metrics.txsGossiped.Add(float64(numTxsGossipped))
```

Comment:
Is this right? This looks like it will re-accumulate previously counted txs into the metric since we're using `numTxsGossiped`. Shouldn't we be using something like `t.metrics.txsGossiped.Add(float64(len(gossipContainer.Txs))` here?